lib/pbench/agent/tool_meister_stop.py

```diff
@@ -65,6 +65,10 @@ def __init__(self, spec: str, benchmark_run_dir: Path, def_host_name: str):
             )
         except FileNotFoundError:
             pass
+        else:
+            # Since this Redis server is locally managed, communication to it
+            # will always be through the "localhost" interface.
```

Comment:
We're assuming that since we have a `redis.pid` in the run directory, we created and manage the local Redis. Which I suppose is a fairly solid assumption, but saying that directly might be better. (That is, justifying the assumption rather than simply stating it.) This is, in any case, consistent with `locally_managed`, which triggers off the lack of a `pid_file` attribute.

Reply:
I have to ask...do we take steps to ensure that there is no `redis.pid` file when we are _not_ orchestrating the Redis server?  (That is, when we start running, do we, e.g., clear out the run directory?...or, could there possibly be a `redis.pid` left over from a previous run?)

[Unresolving.]

Reply:
Each run directory is unique, so there is no previous run to consider.  The `pbench-tool-meister-start` creates the `tm` directory, but Redis is the responsible party for creating the `redis.pid` file.  Only when `pbench-tool-meister-start` is responsible for orchestrating Redis is it directed in the `redis.conf` file to create it at `${benchmark_run_dir}/tm/redis.pid`.

So unless somebody is very clever and creates the expected run directory, `tm` sub-directory, and directs their Redis instance to create the pid file there, we are good.